Add the ability to get/give/drop stacks of things

InspectorCaracal opened the pull request:
#### Brief overview of PR changes/additions
This adds a barebones syntax for pulling the number out of a command argument on get, drop, and give - and more significantly, makes the command funcs therein able to handle moving around multiple objects.

#### Motivation for adding to Evennia
Being able to get, give, and drop multiple of an object is an extremely common and fairly fundamental feature that the vast majority of people using Evennia wind up implementing. I consider it's nearly as fundamental as multimatching, honestly. And since the object search itself supports stacking, it seems reasonable that these basic object manipulation commands should also support stacking.

#### Other info (issues closed, discussion etc)
This doesn't work as smoothly as it should, at the moment, due to the bug described in #3432. Since the pluralization search isn't working correctly, for whatever reason, `get 3 rocks` will only ever get one rock. However, fixing that bug is outside of the scope of this PR, and `get 3 rock` works correctly.

Also, the `get` command had its help file locked to Developer, for some reason, so I fixed that in this PR. If for whatever reason this PR is rejected, I'll put in a smaller patch to fix that separately.

InspectorCaracal commented:
Realized slightly belatedly I should add tests for the stacking syntax, so there you go.

Griatch commented on evennia/commands/default/general.py:
```diff
@@ -387,48 +387,72 @@ class CmdGet(COMMAND_DEFAULT_CLASS):
 
     key = "get"
     aliases = "grab"
-    locks = "cmd:all();view:perm(Developer);read:perm(Developer)"
```
I honestly don't remember why this was :thinking: It seems pretty deliberate though :shrug:

Griatch commented on evennia/commands/default/general.py:
```diff
@@ -387,48 +387,72 @@ class CmdGet(COMMAND_DEFAULT_CLASS):
+    def parse(self):
+        super().parse()
+        self.number = 0
+        if self.args:
+            # check for numbering
+            count, *args = self.args.split(maxsplit=1)
+            # we only use the first word as a count if it's a number and
+            # there is more text afterwards
+            if args and count.isdecimal():
+                self.number = int(count)
+                self.args = args[0]
```
This same block is repeated over and over.
- Maybe commands with item targeting like this should all share the same parent class. 
- Another option would be to add this kind of functionality to the `object.search()`, but possibly that makes it harder to customize.

Griatch reviewed (changes requested):
I think I agree with you that this kind of stack manipulation makes sense to have in core. If deciding to take that step though, it feels like we could help the dev more than to just make custom additions to a few general commands: 

The `obj.search()` method is used in almost all commands. Even today, with `quiet=True`, this can return a list. It already supports custom query input to separate matches (like `apple-4`), so maybe it could be made to also support `two apple` or `2 apple` in a general form (as you say, `two apples` is another issue, but maybe it's time for a general refactor of the search to be more naturalistic ...). I'm open to discussion. 

Alternatively, even with the per-command fixes, I think we could avoid a lot of code repetition here.

Griatch commented:
@InspectorCaracal Ping :)

Griatch commented:
Ping.

InspectorCaracal commented:
I haven't forgotten about this, but I haven't come to a decision about what to do re: your comments. A more fundamental change to searching and/or command targeting *would* be a better solution, but I don't think I'm the right person to tackle it.

Griatch commented:
@InspectorCaracal That's fine. We could move a potential search refactor into a feature request instead. I think we could still get away with less repetition here though.

InspectorCaracal commented:
I've rebased the PR to the current main, pulled out the number parsing portion into a new parent command class, `NumberedTargetCommand`, and also updated the three commands to use the nice new `return_string` keyword in `get_numbered_name`. Once the tests finish, if there's no new issues, it should be good to go :+1:

Griatch reviewed (approved):
Looking cleaner!
